From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mattcc <matt.fr1kin@gmail.com>
Date: Mon, 14 Feb 2022 16:20:58 -0600
Subject: [PATCH] Fix build.gradle.kts


diff --git a/build.gradle.kts b/build.gradle.kts
index 028f6a1795ceb99d1760c73b0980238677b4b8bc..1af86d747e11a335b78dc2b5c730cca9865a55fb 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -18,8 +18,9 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":paper-api"))
-    implementation(project(":paper-mojangapi"))
+//    implementation(project(":PaperX-API"))
+    implementation("io.papermc.paper:paper-api:1.18.1-R0.1-SNAPSHOT")
+    implementation("io.papermc.paper:paper-mojangapi:1.18.1-R0.1-SNAPSHOT")
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.21.0")
     implementation("net.minecrell:terminalconsoleappender:1.3.0")
